File: mysql_examples/recipes/apache/paragraphs/paragraphs.pl
#!/usr/bin/perl
# paragraphs.pl - generate HTML paragraphs

use strict;
use warnings;
use CGI qw(:standard escape escapeHTML);
use Cookbook;

my $title = "Query Output Display - Paragraphs";

print header (), start_html (-title => $title, -bgcolor => "white");

my $dbh = Cookbook::connect ();
my ($now, $version, $user, $db);
my $para;

#@ _DISPLAY_PARAGRAPH_
($now, $version, $user, $db) =
  $dbh->selectrow_array ("SELECT NOW(), VERSION(), USER(), DATABASE()");
$db = "NONE" unless defined ($db);
print p (escapeHTML ("Local time on the MySQL server is $now."));
print p (escapeHTML ("The server version is $version."));
print p (escapeHTML ("The current user is $user."));
print p (escapeHTML ("The default database is $db."));
#@ _DISPLAY_PARAGRAPH_

$dbh->disconnect ();

print end_html ();

File: mysql_examples/recipes/apache/paragraphs/paragraphs.py
#!/usr/bin/python
# paragraphs.py - generate HTML paragraphs

import cgi
import MySQLdb
import Cookbook

title = "Query Output Display - Paragraphs"

# Print content type header, blank line that separates
# headers from page body, and first part of page

print "Content-Type: text/html"
print ""
print "<html>"
print "<head>"
print "<title>" + title + "</title>"
print "</head>"
print "<body bgcolor=\"white\">"

conn = Cookbook.connect ()

#@ _DISPLAY_PARAGRAPH_
cursor = conn.cursor ()
cursor.execute ("SELECT NOW(), VERSION(), USER(), DATABASE()")
(now, version, user, db) = cursor.fetchone ()
cursor.close ()
if db is None:  # check database name
  db = "NONE"
para = ("Local time on the MySQL server is %s.") % now
print "<p>" + cgi.escape (para, 1) + "</p>"
para = ("The server version is %s.") % version
print "<p>" + cgi.escape (para, 1) + "</p>"
para = ("The current user is %s.") % user
print "<p>" + cgi.escape (para, 1) + "</p>"
para = ("The default database is %s.") % db
print "<p>" + cgi.escape (para, 1) + "</p>"
#@ _DISPLAY_PARAGRAPH_

conn.close ()

print "</body>"
print "</html>"
